pkg/chunked/storage_linux.go

```diff
@@ -1141,41 +1141,93 @@ func makeEntriesFlat(mergedEntries []fileMetadata) ([]fileMetadata, error) {
 	return new, nil
 }
 
-func (c *chunkedDiffer) copyAllBlobToFile(destination *os.File) (digest.Digest, error) {
-	var payload io.ReadCloser
-	var streams chan io.ReadCloser
-	var errs chan error
-	var err error
+type streamOrErr struct {
+	stream io.ReadCloser
+	err    error
+}
+
+func getBlobAt(is ImageSourceSeekable, chunksToRequest []ImageSourceChunk, maxStreams int) (chan streamOrErr, error) {
+	streams, errs, err := is.GetBlobAt(chunksToRequest)
+	if err != nil {
+		return nil, err
+	}
+	stream := make(chan streamOrErr)
+	go func() {
+		tooManyStreams := false
+		defer close(stream)
```

Comment:
I’m paranoid about goroutines failing in hard-to-diagnose ways.

```go
go getBlobAtConverterGoroutine()…
…

func getBlobAtConverterGoroutine(stream) {
   err = errors.New("Unexpected error in getBlobAtGoroutine")
   defer func() {
      if err != nil { stream <- err }
      close(err)
   }()
   err = getBlobAtConverter(…)
}
```

would ensure that any panic in the child is caught and reported.